File: Android/OraclePermissionGenerator/src/net/alastairwyse/oraclepermissiongenerator/containers/OracleObjectPermissionSet.java
/*
 * Copyright 2015 Alastair Wyse (http://www.oraclepermissiongenerator.net/oraclepermissiongeneratorandroid/)
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.alastairwyse.oraclepermissiongenerator.containers;

import java.util.*;

/**
 * A set of permissions relating to an object in an Oracle database.
 * @author Alastair Wyse
 */
public class OracleObjectPermissionSet {

    private String objectName;
    private String objectType;
    private String objectOwner;
    private ArrayList<RoleToPermissionMap> objectPermissions;
    private boolean addFlag;
    private boolean removeFlag;
    
    /**
     * @return  The name of the Oracle object that the set of permissions apply to.
     */
    public String getObjectName() {
        return objectName;
    }
    
    /**
     * @param objectName  The name of the Oracle object that the set of permissions apply to.
     */
    public void setObjectName(String objectName) {
        this.objectName = objectName.toUpperCase(Locale.US);
    }
    
    /**
     * @return  The type of the Oracle object.
     */
    public String getObjectType() {
        return objectType;
    }
    
    /**
     * @param objectType  The type of the Oracle object.
     */
    public void setObjectType(String objectType) {
        this.objectType = objectType;
    }
    
    /**
     * @return  The user which owns the Oracle object.
     */
    public String getObjectOwner() {
        return objectOwner;
    }
    
    /**
     * @param objectOwner  The user which owns the Oracle object.
     */
    public void setObjectOwner(String objectOwner) {
        this.objectOwner = objectOwner.toUpperCase(Locale.US);
    }
    
    /**
     * @return  A collection of permissions related to the object.
     */
    public ArrayList<RoleToPermissionMap> getObjectPermissions() {
        return objectPermissions;
    }

    /**
     * @param objectPermissions  A collection of permissions related to the object.
     */
    public void setObjectPermissions(ArrayList<RoleToPermissionMap> objectPermissions) {
        this.objectPermissions = objectPermissions;
    }

    /**
     * @return  Whether add statements are to be included when generating the set of permissions.
     */
    public boolean getAddFlag() {
        return addFlag;
    }

    /**
     * @param  addFlag  Whether add statements are to be included when generating the set of permissions.
     */
    public void setAddFlag(boolean addFlag) {
        this.addFlag = addFlag;
    }

    /**
     * @return  Whether remove statements are to be included when generating the set of permissions.
     */
    public boolean getRemoveFlag() {
        return removeFlag;
    }

    /**
     * @param removeFlag  Whether remove statements are to be included when generating the set of permissions.
     */
    public void setRemoveFlag(boolean removeFlag) {
        this.removeFlag = removeFlag;
    }

    /**
     * Initialises a new instance of the OracleObjectPermissionSet class.
     */
    public OracleObjectPermissionSet() {
        objectPermissions = new ArrayList<RoleToPermissionMap>();
    }

    /**
     * Initialises a new instance of the OracleObjectPermissionSet class.
     * @param objectName  The name of the Oracle object that the set of permissions apply to.
     */
    public OracleObjectPermissionSet(String objectName) {
        this();
        setObjectName(objectName);
    }
}

File: C#/OraclePermissionGeneratorWebServiceAPI.Containers/OracleObjectPermissionSet.cs
/*
 * Copyright 2015 Alastair Wyse (http://www.oraclepermissiongenerator.net/methodinvocationremotingandroid/)
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace OraclePermissionGeneratorWebServiceAPI.Containers
{
    /// <summary>
    /// Container class equivalent to OraclePermissionGeneratorDataModel.OracleObjectPermissionSet to allow serialization using the DataContract attribute.
    /// </summary>
    /// <remarks>Many classes in the OraclePermissionGeneratorDataModel namespace cannot be directly serialized using the DataContract attribute, as they either implement IXmlSerializable or do not define set properties on all data members.</remarks>
    [DataContract]
    public class OracleObjectPermissionSet
    {
        private bool addFlag;
        private bool removeFlag;
        private string objectName;
        private string objectType;
        private string objectOwner;
        private List<RoleToPermissionMap> objectPermissions;

        [DataMember]
        public bool AddFlag
        {
            set
            {
                addFlag = value;
            }
            get
            {
                return addFlag;
            }
        }

        [DataMember]
        public bool RemoveFlag
        {
            set
            {
                removeFlag = value;
            }
            get
            {
                return removeFlag;
            }
        }

        [DataMember]
        public string ObjectName
        {
            set
            {
                objectName = value;
            }
            get
            {
                return objectName;
            }
        }

        [DataMember]
        public string ObjectType
        {
            set
            {
                objectType = value;
            }
            get
            {
                return objectType;
            }
        }

        [DataMember]
        public string ObjectOwner
        {
            set
            {
                objectOwner = value;
            }
            get
            {
                return objectOwner;
            }
        }

        [DataMember]
        public List<RoleToPermissionMap> ObjectPermissions
        {
            set
            {
                objectPermissions = value;
            }
            get
            {
                return objectPermissions;
            }
        }

        public OracleObjectPermissionSet()
        {
        }
    }
}
